c10/core/TensorOptions.h

```diff
@@ -325,14 +325,54 @@ struct C10_API TensorOptions {
 
   // Resolves the ATen backend specified by the current construction axes.
   Backend backend() const noexcept {
-    Backend backend = deviceTypeToBackend(device().type());
+    return at::tensorTypeIdToBackend(computeTensorTypeId());
+  }
+
+  // This is intended to be a centralized location by which we can determine
```

Comment:
Erm... did you mean to copy paste this comment? Cuz it ain't so centralized anymore.